lib/Module/IntrinsicCleaner.cpp

```diff
@@ -49,6 +49,58 @@ bool IntrinsicCleanerPass::runOnModule(Module &M) {
   return dirty;
 }
 
+llvm::Constant* IntrinsicCleanerPass::createConstantVector(llvm::LLVMContext& ctx, 
+  llvm::Value* pConstantVector, unsigned mode) { 
```

Comment:
```suggestion
  llvm::Value& pConstantVector, unsigned mode) { 
```

As you always expect a value for `pConstantVector` provide it by reference.
This eliminates the check later.

Reply:
It should be changed like this